waltz-data/src/main/ddl/liquibase/db.changelog-1.16.xml

```diff
@@ -0,0 +1,171 @@
+                <constraints nullable="false" />
+            </column>
+            <column name="entity_kind"
+                    type="${enum.type}">
+                <constraints nullable="false" />
+            </column>
+            <column name="entity_id"
+                    type="${id.type}">
+                <constraints nullable="false" />
+            </column>
+            <column name="last_updated_at"
+                    type="TIMESTAMP"
+                    defaultValueComputed="${now.value}">
+                <constraints nullable="false" />
+            </column>
+            <column name="last_updated_by"
+                    type="${name.type}">
+                <constraints nullable="false" />
+            </column>
+            <column name="provenance"
+                    type="${provenance.type}">
+                <constraints nullable="false" />
+            </column>
+        </createTable>
+    </changeSet>
+
+    <changeSet id="20190124-3783-2"
+               author="kamransaleem">
+        <comment>3783: Server Usage unique index</comment>
+        <createIndex tableName="server_usage"
+                     unique="true"
+                     indexName="idx_server_usage_server_entity">
+            <column name="server_id"
+                    type="${id.type}" />
+            <column name="entity_kind"
+                    type="${enum.type}" />
+            <column name="entity_id"
+                    type="${id.type}" />
+        </createIndex>
+    </changeSet>
+
+    <changeSet id="20190124-3783-2b"
+                author="kamransaleem">
+        <comment>3783: Server Usage primary key</comment>
+        <addPrimaryKey columnNames="server_id, entity_kind, entity_id"
+                       constraintName="server_usage_pkey"
+                       tableName="server_usage"/>
+    </changeSet>
+
+    <changeSet id="20190124-3783-3"
+               author="kamransaleem">
+        <comment>3783: insert server usages from server information</comment>
+        <sql>
+            INSERT INTO server_usage(server_id, entity_kind, entity_id, last_updated_by, provenance)
+            SELECT
+                min(si2.id) AS server_id,
+                'APPLICATION' AS entity_kind,
+                app.id AS entity_id,
+                'admin' AS last_updated_by,
+                'waltz' AS provenance
+            FROM server_information si1
+            JOIN server_information si2 ON si1.hostname = si2.hostname
+            JOIN application app ON app.asset_code = si1.asset_code
+            GROUP BY
+                si1.hostname,
+                si1.asset_code,
+                app.id
+        </sql>
+    </changeSet>
+
+    <changeSet id="20190124-3783-4"
+               author="kamransaleem">
+        <comment>3783: remove asset code from server information leaving the min id server in each case</comment>
+        <sql>
+            DELETE
+            FROM server_information
+            WHERE id NOT IN (
```

Comment:
Agreed, although this was because MariaDB doesn't like a delete with a select without a select *.  Did this way to avoid one for mariadb and another for MS SQL and Postgres.